pkg/controller/user-cluster-controller-manager/resources/reconciler.go

```diff
@@ -486,6 +487,7 @@ func (r *reconciler) reconcileConfigMaps(ctx context.Context, data reconcileData
 		creators = append(creators, []reconciling.NamedConfigMapCreatorGetter{
 			coredns.ConfigMapCreator(),
 			nodelocaldns.ConfigMapCreator(r.dnsClusterIP),
+			envoyagent.ConfigMapCreator(r.namespace, r.clusterURL.Hostname()),
```

Comment:
Same here, the config for the agent should be included when the tunneling expose strategy is used only.